packages/sdk/src/service_bridge.ts

```diff
@@ -0,0 +1,681 @@
+        this.l2StandardBridge = L2StandardBridge__factory.connect(DEFAULT_L2_CONTRACT_ADDRESSES.L2StandardBridge.toString(), this.l2SignerOrProvider);
+        this.l1Portal = L1Portal__factory.connect(L1PortalAddress, this.l2SignerOrProvider)
+        this.l1Oracle = L1Oracle__factory.connect(DEFAULT_L2_CONTRACT_ADDRESSES.L1Oracle.toString(), this.l2SignerOrProvider);
+        this.l2Portal = L2Portal__factory.connect(DEFAULT_L2_CONTRACT_ADDRESSES.L2Portal.toString(), this.l2SignerOrProvider);
+        this.l2Portal = L2Portal__factory.connect(DEFAULT_L2_CONTRACT_ADDRESSES.L2Portal.toString(), this.l2SignerOrProvider);
+        this.l1Rollup = Rollup__factory.connect(l1RollupAddress, this.l1SignerOrProvider)
+
+    }
+
+    async getL1OracleBlockNumber() {
+        return await this.l1Oracle.number();
+    }
+    /**
+     * Provider connected to the L1 chain.
+     */
+    get l1Provider(): Provider | undefined {
+        if (this.l1SignerOrProvider) {
+            if (Provider.isProvider(this.l1SignerOrProvider)) {
+                return this.l1SignerOrProvider;
+            } else {
+                return this.l1SignerOrProvider.provider;
+            }
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Provider connected to the L2 chain.
+     */
+    get l2Provider(): Provider | undefined {
+        if (this.l2SignerOrProvider) {
+            if (Provider.isProvider(this.l2SignerOrProvider)) {
+                return this.l2SignerOrProvider;
+            } else {
+                return this.l2SignerOrProvider.provider;
+            }
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Signer connected to the L1 chain.
+     */
+    get l1Signer(): Signer {
+        if (Provider.isProvider(this.l1SignerOrProvider)) {
+            throw new Error(`messenger has no L1 signer`)
+        } else {
+            return this.l1SignerOrProvider
+        }
+    }
+
+    /**
+     * Signer connected to the L2 chain.
+     */
+    get l2Signer(): Signer {
+        if (Provider.isProvider(this.l2SignerOrProvider)) {
+            throw new Error(`messenger has no L2 signer`)
+        } else {
+            return this.l2SignerOrProvider
+        }
+    }
+
+
+    /**
+    * Return the status of the deposit message; 
+    * 
+    * @returns MessageStatus 
+    */
+    public async getDepositStatus(bridgeTxLogs: TransactionReceipt): Promise<MessageStatus> {
+
+        const desiredBlockNumber = bridgeTxLogs.blockNumber;
+        let currentOracleBlockNumber = await this.l1Oracle.number();
+
+        const depositEvent = this.l1Portal.interface.parseLog(bridgeTxLogs.logs[1]);
+        const depositHash = depositEvent.args.depositHash;
+
+
+        const depositFinalized = await this.l2Portal.finalizedDeposits(depositHash)
+
+        if (desiredBlockNumber <= currentOracleBlockNumber.toNumber()) {
+            return MessageStatus.READY;
+        } else if (depositFinalized) {
```

Comment:
i think this needs to be checked first, `desiredBlockNumber <= currentOracleBlockNumber.toNumber()` is also true when message status is done

Reply:
you resolved this, but I still think this needs to be fixed